src/main/scala/is/hail/expr/Type.scala

```diff
@@ -33,8 +33,8 @@ object Type {
         genScalar,
         genScalar,
         genScalar,
-        genArb.resize(size - 1).map(TArray),
-        genArb.resize(size - 1).map(TSet),
+        genArb.resize(size - 1).map(TArray.apply),
+        genArb.resize(size - 1).map(TSet.apply),
```

Comment:
You should also generate a boolean for the required bit, modify TArray, etc. `typeCheck` methods to verify the values of the types are non-missing, and modify the TArray.genNonmissingValue to handle the required case.  Note!  genNonmissingValue can generated a missing value!  (We should probably fix/rename that.)  You'll need to filter out missing elements from the element generator.